front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatMargin.ts

```diff
@@ -1,20 +1,40 @@
 import type { Margin } from 'modules/trainschedule/components/ManageTrainSchedule/types';
 import type { PathStep } from 'reducers/osrdconf/types';
 
-const formatMargin = (pathSteps: PathStep[]): Margin => {
+const formatMargin = (pathSteps: PathStep[]): Margin | undefined => {
   const margins: Margin = {
     boundaries: [],
     values: [],
   };
 
   pathSteps.forEach((step, index) => {
+    // for the first step, we add the margin or 'none' if it doesn't exist
     if (index === 0) {
       margins.values.push(step.theoreticalMargin || 'none');
-    } else if (step.theoreticalMargin !== pathSteps[index - 1].theoreticalMargin) {
+    }
+
+    // for the other steps, we add the margin if it's different from the previous one
+    else if (
+      step.theoreticalMargin &&
+      step.theoreticalMargin !== pathSteps[index - 1].theoreticalMargin &&
+      index !== pathSteps.length - 1
+    ) {
       margins.boundaries.push(step.id);
-      margins.values.push(step.theoreticalMargin || 'none');
+      margins.values.push(step.theoreticalMargin);
+    }
+
+    // for the last step, if the origin has a margin and there isn't any via with one,
+    // we add 'none' add the end of the values because we need N+1 values compared to boundaries
+    else if (index === pathSteps.length - 1 && pathSteps[0].theoreticalMargin) {
+      margins.boundaries.push(step.id);
+      margins.values.push('none');
     }
   });
+
+  // check if all margins are 'none'
+  if (margins.values.every((value) => value === 'none')) {
```

Comment:
You can simply check if the `values.length !== 1`
```suggestion
  if (values.length === 1) {
```

Reply:
Done